fe/fe-core/src/main/java/org/apache/doris/nereids/stats/StatsCalculator.java

```diff
@@ -205,6 +210,23 @@ public Map<String, ColumnStatistic> getTotalColumnStatisticMap() {
         return totalColumnStatisticMap;
     }
 
+    /**
+     * disable join reorder if any table row count is not available.
+     */
+    public static void disableJoinReorderIfTableRowCountNotAvailable(
+            List<LogicalOlapScan> scans, CascadesContext context) {
+        StatsCalculator calculator = new StatsCalculator(context);
+        for (LogicalOlapScan scan : scans) {
+            double rowCount = calculator.getOlapTableRowCount(scan);
+            if (rowCount == -1 && ConnectContext.get() != null) {
+                LOG.info("disable join reorder since row count not available: "
+                        + scan.getTable().getNameWithFullQualifiers());
+                ConnectContext.get().getSessionVariable().setDisableJoinReorder(true);
```

Comment:
This sets session variable directly, should use disableNereidsJoinReorderOnce?

Reply:
this is statement level. it has no impact on other queries.